mesonbuild/dependencies/base.py

```diff
@@ -46,6 +46,17 @@ class DependencyException(MesonException):
     '''Exceptions raised while trying to find dependencies'''
 
 
+class MissingCompiler:
+    """Represent a None Compiler - when no tool chain is found.
+    replacing AttributeError with DependencyException"""
+
+    def __getattr__(self, item):
+        raise DependencyException('no toolchain found')
```

Comment:
The testsuite failures happen when python checks if some builtin `__dunder__` methods are defined e.g. to override how classes are pickled (`__getstate__` / `__setstate__` / `__getnewargs__` / `__getnewargs_ex__`). I think the appropriate thing to do here is:

```
if item.startswith('__'):
    raise AttributeError()
raise DependencyException('no toolchain found')
```

So that python can tell that it shouldn't try to handle this class. We only care about raising a DependencyException when trying to call compiler methods.

Reply:
Done.
Sorry for taking so long Its the Jewish holidays.

Reply:
Haha, what a coincidence. That's the same excuse I've been using to procrastinate on reviews. :D

And I've finally gotten the Sukkah down today while the weather wasn't making that difficult. \o/

Reply:
Alright, CI is running...